helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1085,6 +1085,139 @@ public ClusterManagementMode getClusterManagementMode(String clusterName) {
         : new ClusterManagementMode(status.getManagementMode(), status.getManagementModeStatus());
   }
 
+  private enum SetPartitionToErrorFailureReason {
+    INSTANCE_NOT_ALIVE("%s is not alive in cluster %s"),
+    INSTANCE_NON_EXISTENT("%s does not exist in cluster %s"),
+    RESOURCE_NON_EXISTENT("resource %s is not added to cluster %s"),
+    PARTITION_NON_EXISTENT("not all %s exist in cluster %s"),
+    PARTITION_ALREADY_IN_ERROR("%s is NOT found in cluster %s or is already in ERROR state"),
+    STATE_MODEL_NON_EXISTENT("%s is NOT found in cluster %s");
+
+    private String message;
+
+    SetPartitionToErrorFailureReason(String message) {
+      this.message = message;
+    }
+
+    public String getMessage(String resourceName, List<String> partitionNames, String instanceName,
+        String errorStateEntity, String clusterName) {
+      return String.format("Can't set to Error State for %s.%s on %s, because " + message, resourceName,
+          partitionNames, instanceName, errorStateEntity, clusterName);
+    }
+  }
+
+  @Override
+  public void setPartitionToError(String clusterName, String instanceName, String resourceName,
```

Comment:
the code is lot common with 'resetPartition' - can the common code be separated out and add only the deltas ?

Reply:
+1. Please refactor resetPartition if it is needed. Let's leverage the code that already exists.